src/background/popupHandler.js

```diff
@@ -19,7 +20,15 @@ export const showPopup = async (aepp, type, params) => {
   });
 
   const extUrl = browser.runtime.getURL('./index.html');
-  const popupUrl = `${extUrl}?id=${id}&type=${type}&url=${encodeURIComponent(href)}`;
+  let popupUrl;
+  let isRawSign = false;
+  if (type === POPUP_TYPE_SIGN && !isTxOfASupportedType(params?.tx, true)) {
+    isRawSign = true;
+    popupUrl = `${extUrl}?id=${id}&type=${POPUP_TYPE_RAW_SIGN}&url=${encodeURIComponent(href)}`;
+  } else {
+    popupUrl = `${extUrl}?id=${id}&type=${type}&url=${encodeURIComponent(href)}`;
+  }
+
```

Comment:
What are your thoughts of removing that if-else clause?
```suggestion
  const isRawSign = type === POPUP_TYPE_SIGN && !isTxOfASupportedType(params?.tx, true);
  const popupType = isRawSign ? POPUP_TYPE_RAW_SIGN : type
  const popupUrl = `${extUrl}?id=${id}&type=${popupType}&url=${encodeURIComponent(href)}`;
```